File: ios/Classes/FlocationPlugin.m
#import "FlocationPlugin.h"

@implementation FlocationPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  LocationStreamHandler* locationStreamHandler = [[LocationStreamHandler alloc] init];
  FlutterEventChannel* locationChannel = [FlutterEventChannel eventChannelWithName: @"motionmobs.com/flocation" binaryMessenger: [registrar messenger]];
  [locationChannel setStreamHandler: locationStreamHandler];
}
@end

@implementation LocationStreamHandler
- (FlutterError*)onListenWithArguments:(id)arguments eventSink:(FlutterEventSink)eventSink {
    int kNumberOfIterations = 100;

    //TODO: stream location events!
    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
        //Background Thread
        for( int i; i < kNumberOfIterations; i++) {
          NSNumber* lat = [NSNumber numberWithDouble: (double)i];
          NSNumber* lng = [NSNumber numberWithDouble: (double)i];
          NSNumber* elev = [NSNumber numberWithDouble: (double)i];
          NSNumber* acc = [NSNumber numberWithDouble: (double)i];
          NSArray* value = @[lat,lng,elev,acc];
          eventSink(value);
          sleep(1);
        }
      });
    return nil;
}

- (FlutterError*)onCancelWithArguments:(id)arguments {
  //TODO: clean up!
  return nil;
}
@end

File: android/src/main/kotlin/com/yourcompany/flocation/FlocationPlugin.kt
package com.yourcompany.flocation

import android.os.AsyncTask
import java.lang.Thread
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.EventChannel.StreamHandler
import io.flutter.plugin.common.EventChannel.EventSink
import io.flutter.plugin.common.PluginRegistry.Registrar

class FlocationPlugin(): StreamHandler {
  companion object {
    @JvmStatic
    fun registerWith(registrar: Registrar): Unit {
        val LocationID = "motionmobs.com/flocation"
        val channel = EventChannel(registrar.messenger(), LocationID)
      channel.setStreamHandler(FlocationPlugin())
    }
  }



   override fun onListen(arguments: Any?, eventSink: EventSink?){
        //TODO: stream location events!

       TestEvents().execute(eventSink);
    }

   override fun onCancel(arguments: Any?){
        //TODO: clean up!
    }

    inner class TestEvents: AsyncTask<EventSink?, Boolean, Boolean>() {

        override fun onPreExecute() {
            super.onPreExecute()
        }

        override fun doInBackground(vararg events: EventSink?):Boolean {
            for (eventSink in events) {
                if(eventSink != null) {
                    for (i in 0..99) {
                        val lat = i.toDouble();
                        val long = i.toDouble();
                        val elev = i.toDouble();
                        eventSink?.success(listOf(lat, long, elev))
                        Thread.sleep(1_000)
                    }
                    return true
                }
            }
            return false
        }

        override fun onPostExecute(result: Boolean) {
            super.onPostExecute(result)
        }
    }
}
